framework/include/utils/MathFVUtils.h

```diff
@@ -538,45 +558,442 @@ interpolate(const Limiter & limiter,
+    delta_face = (*grad_phi_upwind) * (face_centroid - cell_centroid);
+  else
+    delta_face = (*grad_phi_face) * (face_centroid - cell_centroid);
+
+  // Return the interpolated value
+  return phi_upwind + beta * delta_face;
+}
+
+/**
+ * @brief Computes the minimum and maximum scalar values in a two-cell stencil around a given face
+ * in a computational grid.
+ *
+ * This function calculates the minimum and maximum values within a two-cell stencil. The stencil
+ * includes the immediate neighboring elements of the face's associated element and the neighboring
+ * elements of those neighbors. It evaluates the values using a provided functor and accounts for
+ * the valid (non-null) neighbors.
+ *
+ * @tparam T The data type for the values being computed. This is typically a scalar type.
+ * @tparam FEK Enumeration type FunctorEvaluationKind with a default value of
+ * FunctorEvaluationKind::Value. This determines the kind of evaluation the functor will perform.
+ * @tparam Enable A type trait used for SFINAE (Substitution Failure Is Not An Error) to ensure that
+ * T is a valid scalar type as determined by ScalarTraits<T>.
+ *
+ * @param functor An object of a functor class derived from FunctorBase<T>. This object provides the
+ * genericEvaluate method to compute the value at a given element and time.
+ * @param face An argument representing the face in the computational domain. The face provides
+ * access to neighboring elements via neighbor_ptr_range().
+ * @param time An argument representing the state or time at which the evaluation is performed.
+ *
+ * @return std::pair<T, T> A pair containing the minimum and maximum values computed across the
+ * two-cell stencil. The first element is the maximum value, and the second element is the minimum
+ * value.
+ *
+ * This function performs the following steps:
+ * 1. Initializes max_value to 0 and min_value to a large value (1e10).
+ * 2. Iterates over the direct neighbors of the element associated with the face.
+ *    - If a neighbor is valid (not null), it evaluates the functor at that neighbor and updates
+ * max_value and min_value.
+ * 3. Iterates over the neighbors of the neighbors.
+ *    - Similar to the first loop, it evaluates the functor at valid neighbors and updates max_value
+ * and min_value.
+ * 4. Returns a pair containing the computed max_value and min_value.
+ *
+ * Usage:
+ * This function is typically used in the finite volume methods for min-max computations over
+ * stencils (neighborhoods). It helps compute the limiting for limited second order upwind at the
+ * faces.
+ */
+template <typename T,
+          FunctorEvaluationKind FEK = FunctorEvaluationKind::Value,
+          typename Enable = typename std::enable_if<ScalarTraits<T>::value>::type>
+std::pair<T, T>
+computeMinMaxValue(const FunctorBase<T> & functor, const FaceArg & face, const StateArg & time)
+{
+  // Initialize max_value to 0 and min_value to a large value (1e10)
+  T max_value(0), min_value(1e10);
```

Comment:
advected quantity could go negative if it's momentum ? so a starting max_value of 0 isnt reliable

Reply:
Chaneged by Real bounds. This is a good comment